api/tests/opentrons/protocols/context/simulator/test_protocol_context.py

```
@@ -0,0 +1,80 @@
+"""Test instrument context simulation."""
+import pytest
+from opentrons.protocols.context.protocol import AbstractProtocol
+from pytest_lazyfixture import lazy_fixture
+
+from opentrons.protocols.context.labware import AbstractLabware
+from opentrons.protocols.context.protocol_api.labware import LabwareImplementation
+from opentrons_shared_data.labware.dev_types import LabwareDefinition
+from opentrons import types, ThreadManager
+from opentrons.protocols.context.protocol_api.protocol_context import (
+    ProtocolContextImplementation,
+)
+from opentrons.protocols.context.simulator.protocol_context import (
+    ProtocolContextSimulation,
+)
+
+
+@pytest.fixture
+def protocol_context(hardware: ThreadManager) -> ProtocolContextImplementation:
+    """Protocol context implementation fixture."""
+    return ProtocolContextImplementation(hardware=hardware)
+
+
+@pytest.fixture
+def simulating_protocol_context(hardware: ThreadManager) -> ProtocolContextSimulation:
+    """Protocol context simulation fixture."""
+    return ProtocolContextSimulation(hardware=hardware)
+
+
+@pytest.fixture
+def labware(minimal_labware_def: LabwareDefinition) -> AbstractLabware:
+    """Labware fixture."""
+    return LabwareImplementation(
+        definition=minimal_labware_def,
+        parent=types.Location(types.Point(0, 0, 0), "1"),
+    )
+
+
+@pytest.fixture(
+    params=[
+        lazy_fixture("protocol_context"),
+        lazy_fixture("simulating_protocol_context"),
+    ]
+)
+def subject(request) -> AbstractProtocol:
+    return request.param
+
+
+def test_load_instrument_fails_when_present(subject: AbstractProtocol) -> None:
+    """It should fail if already present."""
+    subject.load_instrument("p300_single_gen2", types.Mount.RIGHT, replace=False)
+    with pytest.raises(RuntimeError):
+        subject.load_instrument("p300_single_gen2", types.Mount.RIGHT, replace=False)
```

Just to confirm my understanding, this should throw an exception because we are not telling it to replace?
This means we are telling it to load a new p300 into the right pipette slot, and we won't allow it because it is already there.

---

That is correct